### PR TITLE
Reduced order eigenstrain as a material.

### PR DESCRIPTION
**Pull request by jasondhales**
This adds a new eigenstrain class whose job it is to take the other eigenstrains and smooth them across each element to match the order of the strain field.  #9560

### CODE REVIEW

**Comment by jasondhales** [comment]
Tagging @bwspenc @sapitts @dschwen

File: modules/tensor_mechanics/include/materials/ComputeReducedOrderEigenstrain.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeEigenstrainBase.h"
+#include "RankTwoTensor.h"
+
+// libmesh includes
+#include "libmesh/dense_matrix.h"
+#include "libmesh/dense_vector.h"
+#include "libmesh/quadrature.h"
```
Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeReducedOrderEigenstrain.h]: the `quadrature.h` include can probably be moved to the `.C` file, right?

File: modules/tensor_mechanics/src/materials/ComputeReducedOrderEigenstrain.C [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+#include "MooseMesh.h"
+
+template <>
+InputParameters
+validParams<ComputeReducedOrderEigenstrain>()
+{
+  InputParameters params = validParams<ComputeEigenstrainBase>();
+  params.addRequiredParam<std::vector<MaterialPropertyName>>(
+      "eigenstrain_names", "List of eigenstrains to be applied in this strain calculation");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/ComputeReducedOrderEigenstrain.C]: `eigenstrain_name` and `eigenstrain_names` for the output and input eigenstrains is a bit of an unfortunate naming convention (I think you took that from the strain calculator). Maybe `input_eigenstrain_names`?

**Comment by dschwen** [comment]
I like this design!

**Comment by permcody** [comment]
@andrsd or @joshuahansel, Did we pick up a bad version of Relap-7 here? I'm puzzled by the error messages.

**Comment by joshuahansel** [comment]
@permcody Yes, you picked up a bad version of RELAP-7. Those errors result from the changes made to Piecewise in #9791. This is fixed in the current RELAP-7.

File: modules/tensor_mechanics/include/materials/ComputeReducedOrderEigenstrain.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  const bool _second_order;
+  /// The sum of all eigenstrains at each integration point
+  std::vector<RankTwoTensor> _eigsum;
+  /// The (num q points x ncols) array for the least squares.  Holds 1, xcoor, ycoor, zcoor.
+  DenseMatrix<Real> _A;
+  /// The b array holding the unique eigenstrain components for each integration point.
+  std::vector<DenseVector<Real>> _b;
+  /// Transpose of A
+  DenseMatrix<Real> _AT;
+  /// Traspose of A times b.
```
Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeReducedOrderEigenstrain.h]: Typo tra**n**spose

**Review by dschwen — DISMISSED** [review]
Looks good. Just a small typo that you might want to fix (or leave for later).

**Comment by permcody** [comment]
@dschwen - Looks like this is ready for a final review and merge.

**Review by dschwen — APPROVED** [review]
👍

**Comment by jasondhales** [comment]
Ping

**Pull request merged**